### PR TITLE
Add test case for Logb function

### PR DESCRIPTION
**Pull request by anshuldata**
* Added logb function test case
* duckdb, Velox, cudf doesn't support it. So skipped adding these dialects
* Also, fixed handling of unsupported_kernel if expected result is an error. Issue observed is as below, 
postgres supports fp64 datatype (i.e. float8 in postgres). But postgres log function doesn't support fp64 arguments. so a test expecting fp64 argument and return type error would fail even if posgres returned error.
* In other words, for unsupported kernel in a function, if test case expected error and dialect returned error then "should_Pass" should be true. But way exising code is, "should_pass" was considered false for unsupported kernel ([here](https://github.com/voltrondata/bft/blob/98d418fa18aec8b3bd1d43f7806e35e9b2013b87/bft/dialects/types.py#L217))

### CODE REVIEW

**Comment by anshuldata** [comment]
@richtia can you check this? I can't add reviewer so CCing

**Comment by anshuldata** [comment]
Hi @westonpace 
Can you check this PR ?

**Review by richtia — APPROVED** [review]
